### ISSUE TITLE
Add option to avoid pod install during "flutter build ios"

### ISSUE THREAD

**Issue by orinzhao** [flutter/flutter#53781]
## Use case

I am trying to setup a flutter iOS build on AppCenter, the project depends on a few company internal pods. I normally commit all installed pods together with the project so never need to do a Pod Install on CI portal.
However, "flutter build ios" will force a Pod Install, have been looking for couple days for ways to disable "Pod Install" but didn't find anything.
Have been trying with Flutter 1.12.13+hotfix.8 • channel stable • and Flutter 1.15.17 • channel beta •

Thanks

## Proposal

Have an option for "flutter build ios" to disable Pod Install, or allow customize Pod command (such as Pod update instead of Pod Install)
Any alternatives are welcome as well.

**Comment by dnfield** [comment]
@jmagman would know

**Comment by jmagman** [comment]
You can edit your [`ios/Podfile`](https://guides.cocoapods.org/syntax/podfile.html) however you want.  
1. Try out `:incremental_installation`.  
1. If you are on v1.15.4 or later ([master, dev, and beta](https://flutter.dev/docs/development/tools/sdk/releases?tab=macos) at the moment), https://github.com/flutter/flutter/pull/51453 will allow you to remove `install! 'cocoapods', :disable_input_output_paths => true`.  (If you need to switch back to an older version of Flutter (back to stable, for example), please follow [these migration instructions](https://flutter.dev/docs/development/ios-project-migration) to make sure you can still build.)
1. Are you checking in your Podfile.lock?

**Comment by orinzhao** [comment]
Great to hear from you, thank you.

I tried to add 1 & 2, shifted to beta
install! 'cocoapods',
incremental_installation: true,
disable_input_output_paths: true
It still doesn't work b/c pod install still try to fetch podspec from company internal git.
  ssh: Could not resolve hostname git.xxxx.com: nodename nor servname provided, or not known
    fatal: Could not read from remote repository.
3. Yes, I check in Podfile.lock.

**Comment by jmagman** [comment]
If Flutter sees `ios/Podfile` it will try to run `pod install` in case any Flutter plugins are in there...
I think you just need to play around with your Podfile.  For example, you could point the pod `path` to your cache instead of the internal git repo.  

pod 'your_pod', :path => 'path/to/cache/your_pod'

**Comment by orinzhao** [comment]
Thank you for the advices. There are workaround for Pods & CI for sure.

Though I am exploring workaround within Flutter tools.
Yes, we are using some Flutter plugins, Pod install is necessary at least for the first time. Though I am wondering if Pod install is necessary for every flutter build. If not, eliminate pod install would resolve my AppCentre issue, and also reduce build time.

Do you mind help confirm if it is possible to disable pod install in flutter build tool, and if not, would it be a plan to include it in the future?

Thanks a lot.

**Comment by jmagman** [comment]
> Do you mind help confirm if it is possible to disable pod install in flutter build tool, and if not, would it be a plan to include it in the future?

I don't think this is in the plans.  How could the tool know that you _don't_ need to update your pods?  The logic about caching is in CocoaPods, not the tool.  If CocoaPods changes that behavior, we don't want the tool to stomp it, or to have different behavior depending on what version of CocoaPods you have installed.  The tool doesn't have a hook to know you changed your Podfile.  It would have to cache somewhere the last time you modified your pubspec, or your Podfile, or know that it didn't run since the last time the Flutter tool itself was updated.

Hopefully by adjusting your Podfile you will be able to reduce the build time, avoid hitting your internal server when the pods are cached, and to avoid rebuilding them when they are.

**Comment by orinzhao** [comment]
Yes, I agree flutter tool shouldn't be managing cocoapods related behavior.
And sorry I guess I didn't make my question clear enough, I actually ask if flutter tool can have a parameter like: "flutter build ios -no-podinstall" and allow developers to manage pod code the way they like.
And of course by default pod install should be enabled.
Thanks again for all your help.

**Comment by jmagman** [comment]
@jonahwilliams pointed out there is already a `--no-pub` flag which does something similar to suppress `pub get` before commands.  We would be willing to look at pull request to add this feature, including tests of course. 😄

**Comment by orinzhao** [comment]
Thanks a lot, just want to share that my test result regarding --no-pub, it does disable "pod install" for the 2nd time build after flutter clean.
If do flutter build right after flutter clean, pod install is not disabled.

**Comment by jmagman** [comment]
> Could Somebody tell me a way to pod update during flutter build?

This is unrelated, please file a new issue.

**Comment by EArminjon** [comment]
Up ?
